### ISSUE TITLE
Masters fail to start if we have a '/' at the end of the state store in the S3 bucket

### ISSUE THREAD

**Issue by kenden** [kubernetes/kops#2439]
If there is a `/` at the end of `KOPS_STATE_STORE`, the masters fail to start with the error:  
`error fetching s3://k8s/kops/k8s.test.redacted.com/cluster.spec: AccessDenied`  
This is because in the IAM role's policy resource, an extra `/` is added.

## Scenario

We created our cluster with: `KOPS_STATE_STORE=s3://k8s/kops/`

The masters of our cluster did not start properly:

After checking the log (in `/var/log/daemon.log`), I found the following:
```
**Apr 26 14:45:40 ip-10-0-107-102 nodeup[736]: I0426 14:45:40.452191     736 s3context.go:114] Found bucket "k8s" in region "eu-west-1"
Apr 26 14:45:40 ip-10-0-107-102 nodeup[736]: I0426 14:45:40.452265     736 s3fs.go:162] Reading file "s3://k8s/kops/k8s.test.redacted.com/cluster.spec"
Apr 26 14:45:40 ip-10-0-107-102 nodeup[736]: W0426 14:45:40.503162     736 main.go:138] got error running nodeup (will retry in 30s): error loading Cluster "s3://k8s/kops/k8s.k8s.test.redacted.com/cluster.spec": error fetching s3://k8s/kops/k8s.test.redacted.com/cluster.spec: AccessDenied: Access Denied**
```

Checking the policy `masters.k8s.test.redacted.com` that got created, I saw a double backslash in the path `'//'`:

```
{
            "Effect": "Allow",
            "Action": [
                "s3:*"
            ],
            "Resource": [
                "arn:aws:s3:::k8s/kops//k8s.test.redacted.com",
                "arn:aws:s3:::k8s/kops//k8s.test.redacted.com/*"
            ]
        },
```

After removing the extra `/`, the masters started properly.
Also, after deleting and recreating with `KOPS_STATE_STORE=s3://k8s/kops`, it worked fine.

## Suggestions:
Error out if the user sets a state file with a `/` at the end of the state store
Or strip the `/`.

## Versions:
$ kops version
Version 1.5.3

Kubernetes 1.5.2

**Comment by geojaz** [comment]
I think we can do both- ie: strip the trailing `/` and leave a note in the logs that we've stripped the trailing slash so users know what's up- and if end up in a weird situation because of stripping the slash (I cannot imagine what that would be) at least users know how they got there.

**Comment by geojaz** [comment]
This should be fixed in both 1.5 and master.
